.github/workflows/build.yml

```diff
@@ -50,7 +50,7 @@ jobs:
     name: Java ${{ matrix.java }}
     env:
       OS: ${{ matrix.os }}
```

Comment:
Only related to change insofar as `matrix.java-version` was also not defined in the matrix context. In this case I think `matrix.os` has been taken from the [matrix examples](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) but is being misused here because it is never set.

I suggest removing this. I can see no usages of `OS` within the codebase, and this is actively confusing. Adding the [filter on the matrix to ensure codecov runs once](https://github.com/awslabs/aws-lambda-powertools-java/pull/1094/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R67) adds to this - if we actually had a 2D matrix with OS as well, which this implies, the filter would not be enough.

Reply:
yep, will try to remove the os to see if it still works.